package_control/clear_directory.py

```diff
@@ -43,6 +65,14 @@ def clear_directory(directory, ignore_paths=None):
         If all of the files and folders were successfully deleted
     """
 
+    if is_directory_symlink(directory) and not ignore_paths:
```

Comment:
So the symlink is only removed if `ignore_paths` is `None` or empty. That might seem a bit weird considering the name of the function but it should be fine as far as I've checked uses of this function.

I guess the alternative would be to do nothing in that case (symlink and empty `ignore_paths`). That would make the semantics more correct.

But whatever I do with this function, I think it will always be a bit unclear the difference between symlink and normal directory handling.